From c5d086a15d32a04416c1a90ead6cf5cc9aadc842 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v2-4.5-c5d086a15d3-20220531T160730Z-avarab@gmail.com>
In-Reply-To: <cover-v2-0.5-00000000000-20220531T160730Z-avarab@gmail.com>
References: <cover-0.5-00000000000-20220521T170939Z-avarab@gmail.com>
	<cover-v2-0.5-00000000000-20220531T160730Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 19 May 2022 10:27:02 +0200
Subject: [PATCH v2 4/5] parse-options.c: use optbug() instead of BUG() "opts"
 check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the assertions added in bf3ff338a25 (parse-options: stop
abusing 'callback' for lowlevel callbacks, 2019-01-27) to use optbug()
instead of BUG(). At this point we're looping over individual options,
so if we encounter any issues we'd like to report the offending option.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 parse-options.c | 31 ++++++++++++++++++++-----------
 1 file changed, 20 insertions(+), 11 deletions(-)

diff --git a/parse-options.c b/parse-options.c
index 78b46ae9698..243016ae30f 100644
--- a/parse-options.c
+++ b/parse-options.c
@@ -473,21 +473,30 @@ static void parse_options_check(const struct option *opts)
 				optbug(opts, "should not accept an argument");
 			break;
 		case OPTION_CALLBACK:
-			if (!opts->callback && !opts->ll_callback)
-				BUG("OPTION_CALLBACK needs one callback");
-			if (opts->callback && opts->ll_callback)
-				BUG("OPTION_CALLBACK can't have two callbacks");
+			if (!opts->callback && !opts->ll_callback) {
+				optbug(opts, "OPTION_CALLBACK needs one callback");
+				break;
+			}
+			if (opts->callback && opts->ll_callback) {
+				optbug(opts, "OPTION_CALLBACK can't have two callbacks");
+				break;
+			}
 			break;
 		case OPTION_LOWLEVEL_CALLBACK:
-			if (!opts->ll_callback)
-				BUG("OPTION_LOWLEVEL_CALLBACK needs a callback");
-			if (opts->callback)
-				BUG("OPTION_LOWLEVEL_CALLBACK needs no high level callback");
+			if (!opts->ll_callback) {
+				optbug(opts, "OPTION_LOWLEVEL_CALLBACK needs a callback");
+				break;
+			}
+			if (opts->callback) {
+				optbug(opts, "OPTION_LOWLEVEL_CALLBACK needs no high level callback");
+				break;
+			}
 			break;
 		case OPTION_ALIAS:
-			BUG("OPT_ALIAS() should not remain at this point. "
-			    "Are you using parse_options_step() directly?\n"
-			    "That case is not supported yet.");
+			optbug(opts, "OPT_ALIAS() should not remain at this point. "
+			       "Are you using parse_options_step() directly?\n"
+			       "That case is not supported yet.");
+			break;
 		default:
 			; /* ok. (usually accepts an argument) */
 		}
-- 
2.36.1.1100.g16130010d07

